pkg/ccl/backupccl/backuprand/backup_rand_test.go

```diff
@@ -178,17 +178,27 @@ database_name = 'rand' AND schema_name = 'public'`)
 		sqlDB.Exec(t, "DROP DATABASE IF EXISTS restoredb")
 		sqlDB.Exec(t, "CREATE DATABASE restoredb")
 
-		tableQuery := fmt.Sprintf("RESTORE rand.* FROM LATEST IN $1 WITH OPTIONS (into_db='restoredb'%s%s)", runSchemaOnlyExtension, withOnlineRestore())
-		if err := backuptestutils.VerifyBackupRestoreStatementResult(
+		online := withOnlineRestore()
+		tableQuery := fmt.Sprintf("RESTORE rand.* FROM LATEST IN $1 WITH OPTIONS (into_db='restoredb'%s%s)", runSchemaOnlyExtension, online)
+		verifyStatementResult := backuptestutils.VerifyBackupRestoreStatementResult
+		if online != "" {
+			verifyStatementResult = backuptestutils.VerifyOnlineRestoreStatementResult
+		}
```

Comment:
I'm not particularly happy about this aesthetically, any suggestions?

`VerifyBackupRestoreStatementResult` applies to both backup and restore statements, which is fine for regular backup and restore since they both share the same output structure, but doesn't work for online restore.

I did have an idea of creating a general `VerifyBackupRestoreStatementResult` that does a switch between the two current verification functions and would take an additional boolean `isOnline` argument, but the switch seemed like it'd be a little ugly since there is only separate logic for online restores specifically, so I ditched that idea.